lib/ttfunk/table/cff/charstring.rb

```diff
@@ -0,0 +1,489 @@
+module TTFunk
+  class Table
+    class Cff < TTFunk::Table
+      class Charstring
+        CODE_MAP = {
+          1  => :hstem,
+          3  => :vstem,
+          4  => :vmoveto,
+          5  => :rlineto,
+          6  => :hlineto,
+          7  => :vlineto,
+          8  => :rrcurveto,
+          10 => :callsubr,
+          12 => :flex_select,
+          14 => :endchar,
+          18 => :hstemhm,
+          19 => :hintmask,
+          20 => :cntrmask,
+          21 => :rmoveto,
+          22 => :hmoveto,
+          23 => :vstemhm,
+          24 => :rcurveline,
+          25 => :rlinecurve,
+          26 => :vvcurveto,
+          27 => :hhcurveto,
+          28 => :shortint,
+          29 => :callgsubr,
+          30 => :vhcurveto,
+          31 => :hvcurveto
+        }.freeze
+
+        FLEX_CODE_MAP = {
+          35 => :flex,
+          34 => :hflex,
+          36 => :hflex1,
+          37 => :flex1
+        }.freeze
+
+        attr_reader :glyph_id, :raw
+
+        def initialize(glyph_id, top_dict, font_dict, raw)
+          @glyph_id = glyph_id
+          @top_dict = top_dict
+          @font_dict = font_dict
+          @raw = raw
+
+          @default_width_x = (@font_dict || @top_dict)
```

Comment:
Ok, so the `Charstring` class commits a fair number of heinous ruby crimes in order to maximize performance. I was able to dramatically decrease the time it takes to subset the Noto Sans font (tens of thousands of glyphs) by using ivars instead of `attr_readers`. The function call overhead really makes a difference.

Reply:
In this particular instance I meant it can be a local variable since it doesn't seem to be used outside of this method.

Reply:
Oh hehe ok 👍

Reply:
Could you please address this one?

Reply:
Oops sorry. It's done now.

Reply:
Can't see it. Have you forgot to push the change?

Reply:
Lol yes I did. It's pushed now.